Display LED ON, LED OFF.

Surya-Santhosh opened the pull request:
Modify the existing TimeApp program and add a feature to simulate an infinite printf of "LED ON" "LED OFF"with a 1-second delay.

Surya-Santhosh committed:
Create new branch name cross_compilation

Surya-Santhosh committed:
add a new function to print UTC,IST,PST in appTimer.c.
Added a feature print LED ON and LED OFF to main function.

Surya-Santhosh committed:
Change AppTimerDisplayTimeDate() from global to static.

Surya-Santhosh committed:
Remove the lines in header block in main.c and main.c

Surya-Santhosh committed:
Change format specifier unsigned long from %ld to %lu.
Chnage if condition.

Surya-Santhosh committed:
Added appTimerbinary(binary executable file) file.
To ignore binary executable file added appTimerbinary in .gitignore.
Remove line before }.

adarshtrenser2023 commented on main.c:
```diff
@@ -1,95 +1,75 @@
+            printf("Epoch Time is Zero.");
+        }
+
+        printf("\nLED ON\n");
+
+        sleep(1);
+        
+        printf("\x1b[H"); // Move Cursor to top-left
+        printf("\x1b[J"); // clear screen
+
+        ulEpoch = time(&ulTime);
+
+        // Print date and time in UTC,IST,PST. 
+        if (false == AppTimer(ulEpoch))
+        {
+            printf("Epoch Time is Zero.");
+        }
+
+        printf("\nLED OFF\n");
+
+        sleep(1);
+        
+        printf("\x1b[H"); // Move Cursor to top-left
```
why there is 2 sleep
LED ON and OFF should be a fucntion that toggles every 1 sec not a direct print statement.
also need LED blink in a different folder structure
there are binaries in repo why git ignore is not properly working.

Surya-Santhosh committed:
Added new folder LED Display it contain LEDDisplay.h and LEDDisplay.c to display LED Status.
Modified main.c.
Modified .gitignore file.
Remove Coredump files.
Modified Readme file.

Surya-Santhosh committed:
Add executable binary file.

Surya-Santhosh committed:
Corrected variable name - pucLedStatus

Surya-Santhosh committed:
Updated code as per the review.
Used static and remove pointer.

adarshtrenser2023 commented on LEDDisplay/LEDDisplay.c:
```diff
@@ -0,0 +1,60 @@
+//***************************** Local Constants ********************************
+
+//***************************** Local Variables ********************************
+static uint8 sucLedStatus = 0;
```
this is global , we dont need global here

adarshtrenser2023 commented on LEDDisplay/LEDDisplay.c:
```diff
@@ -0,0 +1,60 @@
+{
+    bool blResult = true;
+
+    if (CHECK_ONE < sucLedStatus)
```
what is CHECK_ONE - already told MACRO name should be a name that should imply what it does
eg: here if you are checking status of LED then it can be named CHECK_LED_STATUS
even though for this program i don't think this is needed.
also why the ucLedStatus is uint8 type?

adarshtrenser2023 commented on LEDDisplay/LEDDisplay.h:
```diff
@@ -0,0 +1,34 @@
+//***************************** Global Constants ******************************* 
+#define LED_ON             (1)
+#define LED_OFF            (0)
+#define CHECK_ONE          (1)
```
You can implement the code without this macros

adarshtrenser2023 commented on LEDDisplay/LEDDisplay.c:
```diff
@@ -0,0 +1,60 @@
+            sucLedStatus = LED_OFF;
+        }
+
+        blResult = true;
```
Already true then why again this statement

adarshtrenser2023 commented on main.c:
```diff
@@ -1,95 +1,64 @@
+        }
+
+        sleep(1);
+        printf("\x1b[H"); // Move Cursor to top-left
```
New line missing after sleep

adarshtrenser2023 commented on main.c:
```diff
@@ -1,95 +1,64 @@
+        // To print LED ON, LED OFF.
+        if (false == LEDStatus())
+        {
+            printf("Invalid");
```
what is invalid here ?

adarshtrenser2023 commented on LEDDisplay/LEDDisplay.c:
```diff
@@ -0,0 +1,60 @@
+//******************************************************************************
+
+//******************************* Include Files ********************************
+#include "common.h"
```
common.h is already included in LEDDisplay.h

adarshtrenser2023 commented on LEDDisplay/LEDDisplay.h:
```diff
@@ -0,0 +1,34 @@
+// All Rights Reserved 
+//****************************************************************************** 
+//
+// Summary : Display LED ON, LED OFF with a 1-second delay.
```
1 sec delay is done in main,c note in LED Display.c or .h
since this is .h it should contain something like This File contains function prototypes and macro definition nedded for LED Display

adarshtrenser2023 commented on LEDDisplay/LEDDisplay.c:
```diff
@@ -0,0 +1,60 @@
+// All Rights Reserved
+//******************************************************************************
+// File    : LEDDisplay.c
+// Summary : Display LED ON, LED OFF.
```
Display LED ON and OFF will be better instead of ","

adarshtrenser2023 commented on LEDDisplay/LEDDisplay.c:
```diff
@@ -0,0 +1,60 @@
+//******************************** AppTimer ************************************
```
APPTImer?

adarshtrenser2023 commented on LEDDisplay/LEDDisplay.h:
```diff
@@ -0,0 +1,34 @@
+//******************************* AppTimer *************************************
```
AppTimer?

adarshtrenser2023 commented on LEDDisplay/LEDDisplay.h:
```diff
@@ -0,0 +1,34 @@
+//*********************** Inline Method Implementations ************************ 
+
+#endif 
+// _APPTIMER_H_ 
```
...

adarshtrenser2023 commented on main.c:
```diff
@@ -1,95 +1,64 @@
-    return 0;
-}
-
+//******************************* AppTimer *************************************
```
Need to update now this main is Generic , so replace with main

adarshtrenser2023 commented on common.h:
```diff
@@ -1,30 +1,30 @@
-//*********************** Inline Method Implementations ************************ 
-
-#endif // _COMMON_H_ 
+//******************************8*** TimerApp **********************************
```
rename as Common

Surya-Santhosh committed:
Merge branch 'main' into cross_compilation

Surya-Santhosh committed:
Updated code as per the review.

adarshtrenser2023 commented on appTimer/appTimer.c:
```diff
@@ -1,161 +1,223 @@
+// Return  : blResult
+// Notes   : None
+//******************************************************************************
+bool AppTimerDisplayUTCISTPST(uint32 ulEpoch)
```
not a good naming for function insted you can name StandardTimeConverter something like that

Surya-Santhosh committed:
Change function name -
AppTimerDisplayUTCISTPST to AppTimerTimeConverter
